packages/apollo-server-core/src/requestPipelineAPI.ts

```diff
@@ -57,6 +67,8 @@ export interface GraphQLRequestContext<TContext = Record<string, any>> {
   readonly operationName?: string | null;
   readonly operation?: OperationDefinitionNode;
 
+  readonly overallCachePolicy?: Required<CacheHint> | undefined;
```

Comment:
Because `CacheHint`, `CacheScope`, and `overallCachePolicy` are tied to our caching spec, it might be better to leave them in `apollo-cache-control`. We can still augment `GraphQLRequestContext` with `overallCachePolicy` there.

Reply:
What do you mean by augment? Is that a TypeScript feature I'm not familiar with? I think I'm not following this suggestion.

Reply:
Yep, it's called 'module augmentation' and leads to merging of the declarations: https://www.typescriptlang.org/docs/handbook/declaration-merging.html

So you should be able to add something like this to `apollo-cache-control`:
```
declare module 'apollo-server-core/dist/requestPipelineAPI' {
  interface GraphQLRequestContext<TContext> {
    readonly overallCachePolicy?: Required<CacheHint> | undefined;
  }
}
```

As a side note, we've been thinking about extracting `requestPipelineAPI` into its own `apollo-server-api` package to avoid circular dependencies.